lib/segment/src/problems/unindexed_field.rs

```diff
@@ -0,0 +1,352 @@
+use std::any::TypeId;
+use std::collections::{HashMap, HashSet};
+use std::sync::OnceLock;
+use std::time::Duration;
+
+use http::{HeaderMap, Method, Uri};
+use issues::{Action, Code, ImmediateSolution, Issue, Solution};
+use itertools::Itertools;
+use strum::IntoEnumIterator as _;
+
+use crate::common::operation_error::OperationError;
+use crate::data_types::text_index::{TextIndexParams, TextIndexType, TokenizerType};
+use crate::json_path::{JsonPathInterface, JsonPathV2};
+use crate::types::{
+    AnyVariants, Condition, FieldCondition, Filter, Match, MatchValue, PayloadFieldSchema,
+    PayloadKeyType, PayloadSchemaParams, PayloadSchemaType, RangeInterface,
+};
+#[derive(Debug)]
+pub struct UnindexedField {
+    field_name: JsonPathV2,
+    field_schemas: HashSet<PayloadFieldSchema>,
+    collection_name: String,
+    endpoint: Uri,
+    instance_id: String,
+}
+
+impl UnindexedField {
+    pub fn slow_search_threshold() -> Duration {
+        static SLOW_SEARCH_THRESHOLD: OnceLock<Duration> = OnceLock::new();
+
+        *SLOW_SEARCH_THRESHOLD.get_or_init(|| {
+            Duration::from_secs_f32(
+                std::env::var("QDRANT_SLOW_SEARCH_THRESHOLD")
```

Comment:
the env. variable does not specify the expected unit

Reply:
-> `QDRANT_SLOW_SEARCH_SECS`?